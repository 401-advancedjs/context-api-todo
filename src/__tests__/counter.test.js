import React from 'react';
import Counter from '../components/counter/counter.js';
import RefactorContext from '../components/context/context.js';
import renderer from 'react-test-renderer';

describe('<Counter />', () => {
  test('basic rendering', () => {
    const mountedCounter = shallow(<Counter />);
    expect(mountedCounter.find('h2')).toBeTruthy();
  });

  test('handles increments', () => {
    const mountedContext = mount(<RefactorContext />);
    const form = mountedContext.find('form');

    form.simulate('submit');
    expect(mountedContext.state('count').toEqual(1));
  });

  test('rendering follows the snapshot', () => {
    const snapshot = renderer.create(<Counter />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});