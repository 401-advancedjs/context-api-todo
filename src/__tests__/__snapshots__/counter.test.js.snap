// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> rendering follows the snapshot 1`] = `
<h2>
  There are 
   items in the list
</h2>
`;
